webApp/server_modules/routes/profile.js

```diff
@@ -149,10 +156,47 @@ router.post('/updateProfile', function(req, res) {
     updateUser(req.user, req.body)
         .then(() => res.json({ sucessful: true }))
         .catch(() => res.json({ sucessful: false }));
+
```

Comment:
The same problem appears here with `res.json()` executing before they are added or removed from the courses as a tutor